src/credential/issuance/03-start-credential-issuance.ts

```diff
@@ -339,13 +343,18 @@ const authorizeUserWithQueryMode = async (
   const authRedirectUrl = await authorizationContext
     .authorize(`${authzRequestEndpoint}?${params}`, redirectSchema)
     .catch((e) => {
-      throw new IdentificationError(e.message);
+      throw new AuthorizationError(e.message);
     });
 
   const urlParse = parseUrl(authRedirectUrl);
   const authRes = AuthorizationResultShape.safeParse(urlParse.query);
   if (!authRes.success) {
-    throw new IdentificationError(authRes.error.message);
+    const authErr = AuthorizationErrorShape.safeParse(urlParse.query);
+    if (!authErr.success) throw new AuthorizationError(authRes.error.message); // an error occured while parsing the result and the error
```

Comment:
What do you think to add brackets here? 
I think it's more readable and sticks to the style of the rest of the code.

```suggestion
    if (!authErr.success) {
        throw new AuthorizationError(authRes.error.message); // an error occured while parsing the result and the error
    }
```

Reply:
Done, thanks.